File: src/main/java/dev/elysium/eraces/bootstrap/LoggerConfigurator.java
package dev.elysium.eraces.bootstrap;

import dev.elysium.eraces.ERaces;
import dev.elysium.eraces.exceptions.InitFailedException;

import java.util.logging.Level;

public class LoggerConfigurator implements IInitializer {
    @Override
    public void setup(ERaces plugin) {
        try {
            if (plugin.getContext().getGlobalConfigManager().getData().isDebug()) {
                plugin.getLogger().setLevel(Level.FINE);
                plugin.getLogger().info("Режим отладки включён.");
            }
        } catch (Exception e) {
            throw new InitFailedException("Ошибка при настройке логирования", e);
        }
    }
}

File: src/main/java/dev/elysium/eraces/bootstrap/LoggerConfigurator.kt
package dev.elysium.eraces.bootstrap

import dev.elysium.eraces.ERaces
import dev.elysium.eraces.ERaces.Companion.logger
import dev.elysium.eraces.exceptions.internal.InitFailedException
import java.util.logging.Level

class LoggerConfigurator : IInitializer {
    override fun setup(plugin: ERaces) {
        try {
            if (plugin.context.globalConfigManager.getData().isDebug()) {
                logger().level = Level.FINE
                logger().info("Режим отладки включён.")
            }
        } catch (e: Exception) {
            throw InitFailedException("Ошибка при настройке логирования", e)
        }
    }
}
